Add resource for database

Permagate opened the pull request:
Implement `digitalocean_database` as per #222.

What do you think about the resource name? In API and godo, `Database` represents the database cluster while `DB` represents the database itself.

### Example Usage

```hcl
resource "digitalocean_database" "database-example" {
  cluster_id = "${digitalocean_database_cluster.postgres-example.id}"
  name       = "foobar"
}

resource "digitalocean_database_cluster" "postgres-example" {
  name       = "example-postgres-cluster"
  engine     = "pg"
  version    = "11"
  size       = "db-s-1vcpu-1gb"
  region     = "nyc1"
  node_count = 1
}
```

andrewsomething commented:
Again, awesome work! Thank so much for taking these on.

> What do you think about the resource name? In API and godo, Database represents the database cluster while DB represents the database itself.

While a bit redundant, it would be more consistent with other DigitalOcean tooling, e.g. `doctl databases db <get|list|create|delete>`

```
resource "digitalocean_database_db" "database-example" {
  cluster_id = "${digitalocean_database_cluster.postgres-example.id}"
  name       = "foobar"
}
```

Permagate commented:
@andrewsomething Agreed. I have updated to use `digitalocean_database_db` name instead.

Also added a sneaky typo fix because of copy-paste in import test for `digitalocean_database_user` resource, sorry 😅

andrewsomething commented:
:shipit:  Thanks again!
